common/include/common/base_agent_interface.h

```diff
@@ -21,6 +21,7 @@ class base_agent_interface
 	explicit base_agent_interface(index_t index) : index(index) {}
 
 	virtual std::span<real_t> position() = 0;
+	index_t get_index() const { return index; }
```

Comment:
it is a bad practice to go against the architectural decisions just for the sake of testing. 
I see that you are using this only in a test and only to see whether the index increments - thats not worth adding a new function to the interface